credentials/alts/internal/handshaker/handshaker.go

```diff
@@ -60,8 +60,7 @@ var (
 		},
 	}
 	// control number of concurrent created (but not closed) handshakers.
-	mu                   sync.Mutex
-	concurrentHandshakes = int64(0)
+	sem *semaphore.Weighted
 	// errDropped occurs when maxPendingHandshakes is reached.
 	errDropped = errors.New("maximum number of concurrent ALTS handshakes is reached")
```

Comment:
The test is checking for `err != errDropped`, does that need to change?

Also, is the `err != errDropped` check unnecessary? Because the test didn't fail after this change.

Reply:
This is because `sem.Acquire(ctx, 1)` is always returning a `nil` error and I think the reason is because when `n > s.size` (where `n` is the weight we're trying to acquire and `s.size` is the max allowed weight), `Acquire` returns `ctx.Err()` [1] which I think should be nil for a background context.

Therefore, I reverted the code to use the simple local implementation of acquire and release but without the `n` argument.

[1] https://github.com/golang/sync/blob/master/semaphore/semaphore.go#L52

Reply:
Do you want it to wait?

https://godoc.org/golang.org/x/sync/semaphore#Weighted.TryAcquire should do the same thing as the simple implementation.

But also, if you don't need the wait acquire, this simple implementation should be good enough.

Reply:
Yes, I don't need it to wait this is why I reverted the code to the simple implementation :)